aws/resource_aws_efs_access_point_test.go

```diff
@@ -0,0 +1,515 @@
+					resource.TestCheckResourceAttr(resourceName, "posix_user.0.secondary_gids.#", "1")),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSEFSAccessPoint_tags(t *testing.T) {
+	var ap efs.AccessPointDescription
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_efs_access_point.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEfsAccessPointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSAccessPointConfigTags1(rName, "key1", "value1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEFSAccessPointConfigTags2(rName, "key1", "value1updated", "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+			{
+				Config: testAccAWSEFSAccessPointConfigTags1(rName, "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSEFSAccessPoint_disappears(t *testing.T) {
+	var ap efs.AccessPointDescription
+	resourceName := "aws_efs_access_point.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVpnGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSAccessPointConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					testAccAWSEFSAccessPointDisappears(&ap),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckEfsAccessPointDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).efsconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_efs_access_point" {
+			continue
+		}
+
+		resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+			AccessPointId: aws.String(rs.Primary.ID),
+		})
+		if err != nil {
+			if isAWSErr(err, efs.ErrCodeAccessPointNotFound, "") {
+				return nil
```

Comment:
This loop should `continue` so other `aws_efs_access_point` are checked as well

```suggestion
				continue
```